src/components/UserProfile.js

```diff
@@ -0,0 +1,57 @@
+import React, { useState, useEffect } from 'react';
+import firebase from 'firebase/app';
+import 'firebase/auth';
+import styled from '@emotion/styled';
+import GoogleLoginButton from './GoogleLoginButton';
+
+function UserProfile() {
+  const [userInfo, setUserInfo] = useState(null);
+  const [isLogin, setIsLogin] = useState(false);
+
+  useEffect(() => {
+    if (!userInfo) return;
+
+    const sessionStorage = window.sessionStorage;
+
+    sessionStorage.setItem('userInfo', userInfo);
+    setIsLogin(true);
```

Comment:
`isLogin` 상태를 없앤다면 이렇게 바꿀 수 있겠네요. sessionStorage도 굳이 변수로 뺄 필욘 없어보여요.

```suggestion
    window.sessionStorage.setItem('userInfo', userInfo);
```